dotCMS/src/integration-test/java/com/dotcms/csspreproc/CSSPreProcessServletTest.java

```diff
@@ -0,0 +1,249 @@
+            Assert.fail(String.format("An error occurred when compiling the test SCSS file '%s'. Aborting test " +
+                                              "execution...", inputScssFile));
+        }
+        // Assertions
+        if (expectError) {
+            Assert.assertTrue("The SASS compilation must have failed with an error message.", cssCode.startsWith(
+                    "/* Error:"));
+        } else {
+            Assert.assertTrue("There must be compiled CSS code in the response.", UtilMethods.isSet(cssCode));
+        }
+        Assert.assertNotEquals("The SCSS file could not be found.", HttpStatus.SC_NOT_FOUND, status);
+        Assert.assertNotEquals("The SCSS file could not be read by the specified dotCMS user.",
+                HttpStatus.SC_FORBIDDEN, status);
+        Assert.assertNotEquals("The SASS Compiler output is null.", HttpStatus.SC_INTERNAL_SERVER_ERROR, status);
+    }
+
+    /**
+     * Mocks all the required objects so that the {@link CSSPreProcessServlet} can run without problems.
+     *
+     * @param mockRequest             Mocked request.
+     * @param mockResponse            Mocked response.
+     * @param mockSession             Mocked session.
+     * @param mockServletOutputStream Mocked Servlet Output Stream used to retrieve the actual response.
+     * @param inputScssFile           Test SCSS file that will be compiled.
+     *
+     * @throws DotDataException     An error occurred when retrieving data form dotCMS.
+     * @throws DotSecurityException The specified User doesn't have the required permissions to perform this operation.
+     * @throws IOException          Servlet output and/or writer objects failed.
+     */
+    private void hydrateMockedObjects(final HttpServletRequest mockRequest, final HttpServletResponse mockResponse,
+                                      final HttpSession mockSession,
+                                      final MockServletOutputStream mockServletOutputStream,
+                                      final FileAsset inputScssFile) throws DotDataException, DotSecurityException, IOException {
+        when(mockRequest.getParameter("host_id")).thenReturn(defaultSiteId);
+        when(mockRequest.getParameter("recompile")).thenReturn("true");
+        when(mockRequest.getRequestURI()).thenReturn(inputScssFile.getURI());
+        when(mockRequest.getAttribute(com.liferay.portal.util.WebKeys.USER)).thenReturn(APILocator.getUserAPI().loadUserById("dotcms.org.1"));
+        when(mockRequest.getSession(false)).thenReturn(mockSession);
+        when(mockResponse.getOutputStream()).thenReturn(mockServletOutputStream);
+        when(mockResponse.getWriter()).thenReturn(new PrintWriter(mockServletOutputStream));
+        when(mockSession.getAttribute(WebKeys.CMS_SELECTED_HOST_ID)).thenReturn(defaultSiteId);
+    }
+
+    /**
+     * Mock ServletOutputStream for testing purposes only.
+     */
+    private class MockServletOutputStream extends ServletOutputStream {
```

Comment:
Just curious if we have something like this implemented for some other test. If so, it might be a good idea of refactoring to its own public class and reuse it accordinly.

Reply:
Done!